modules/validations/outdated_tags.js

```diff
@@ -17,58 +16,126 @@ let _nsi;
+    .then(() => locationManager.mergeLocationSets([]) );
+  }
+
+  // Fetch the name-suggestion-index data
+  waitForNSIPresets()
+    .then(() => Promise.all([
+      fileFetcher.get('nsi_data'),
+      fileFetcher.get('nsi_replacements'),
+      fileFetcher.get('nsi_trees')
+    ]))
+    .then(vals => {
+      if (_nsi) return _nsi;
+
       _nsi = {
-        brands: d.brands,
-        matcher: matcher(),
-        wikidata: {},
-        wikipedia: {}
+        data:          vals[0].nsi,            // the raw name-suggestion-index data
+        replacements:  vals[1].replacements,   // trivial old->new qid replacements
+        trees:         vals[2].trees,          // metadata about trees, main tags
+        keys:          new Set(),              // primary osm keys to check for a NSI match
+        qids:          new Map(),              // Map wd/wp tag values -> qids
+        ids:           new Map()               // Map id -> NSI item
       };
 
-      // initialize name-suggestion-index matcher
-      _nsi.matcher.buildMatchIndex(d.brands);
+      _nsi.matcher = Matcher();
+      _nsi.matcher.buildMatchIndex(_nsi.data);
+      _nsi.matcher.buildLocationIndex(_nsi.data, locationManager.loco());
+
+      Object.keys(_nsi.data).forEach(tkv => {
+        const parts = tkv.split('/', 3);     // tkv = "tree/key/value"
+        const t = parts[0];
+        const k = parts[1];
+
+        // Collect primary keys  (e.g. "amenity", "craft", "shop", "man_made", "route", etc)
+        _nsi.keys.add(k);
+
+        const tree = _nsi.trees[t];     // e.g. "brands", "operators"
+        const mainTag = tree.mainTag;   // e.g. "brand:wikidata", "operator:wikidata", etc
 
-      // index all known wikipedia and wikidata tags
-      Object.keys(d.brands).forEach(kvnd => {
-        const brand = d.brands[kvnd];
-        const wd = brand.tags['brand:wikidata'];
-        const wp = brand.tags['brand:wikipedia'];
-        if (wd) { _nsi.wikidata[wd] = kvnd; }
-        if (wp) { _nsi.wikipedia[wp] = kvnd; }
+        const items = _nsi.data[tkv] || [];
+        items.forEach(item => {
+          // Cache NSI ids and main tags
+          item.mainTag = mainTag;
+          _nsi.ids.set(item.id, item);
+
+          // Cache Wikidata/Wikipedia values, for #6416
+          const wd = item.tags[mainTag];
+          const wp = item.tags[mainTag.replace('wikidata', 'wikipedia')];
+          if (wd)         _nsi.qids.set(wd, wd);
+          if (wp && wd)   _nsi.qids.set(wp, wd);
+        });
       });
 
+      _nsi.keys.add('building');  // fallback can match building=* for some categories
+
       return _nsi;
     })
     .catch(() => { /* ignore */ });
 
 
+  // Returns true if this tag key is a "namelike" tag that the NSI matcher would have indexed..
+  function isNamelike(k) {
+    const namePatterns = [
+      /^(flag:)?name$/i,                                             // e.g. `name`, `flag:name`
```

Comment:
openstreetmap/id-tagging-schema#114 requests a `flag:name` field for the Flagpole preset.